### PR TITLE
python-rados: implement new aio_stat.

### PR DESCRIPTION
**Pull request by iain-buclaw-sociomantic**
This change adds python API for the following C methods:
* rados_aio_stat

Signed-off-by: Iain Buclaw <iain.buclaw@sociomantic.com>

### CODE REVIEW

**Comment by iain-buclaw-sociomantic** [comment]
Would this suffer from the same problem as `aio_read`?

**Comment by jdurgin** [comment]
this looks good, could you add a simple test for it to src/test/pybind/test_rados.py?

What problem with aio_read are you encountering?

**Comment by iain-buclaw-sociomantic** [comment]
> What problem with aio_read are you encountering?

I see that with aio_read, it stores the buffer inside the python completion.  Just wondering if the same would be required for aio_stat too.

https://github.com/ceph/ceph/blob/master/src/pybind/rados/rados.pyx#L2125-L2130

**Comment by iain-buclaw-sociomantic** [comment]
Rebased, previous HEAD was https://github.com/ceph/ceph/pull/11006/commits/41e3e0085c37b6acf463b4567ce7d0a485393c59

File: src/test/pybind/test_rados.py [hunk 1]
```diff
@@ -672,6 +672,32 @@ def cb(blah):
...
+                lock.notify()
+
+        comp = self.ioctx.aio_stat("foo", cb)
+        comp.wait_for_complete()
+        with lock:
+            while count[0] < 1:
+                lock.wait()
+        eq(comp.get_return_value(), -2)
+
+        self.ioctx.write("foo", "bar")
```
Comment (jdurgin) [inline src/test/pybind/test_rados.py]: the data written should be a bytes object for python 3

**Comment by jdurgin** [comment]
I'm not certain whether it's required, but it's more clearly correct to explicitly include the reference to the time and size in the completion object

**Comment by iain-buclaw-sociomantic** [comment]
> I'm not certain whether it's required, but it's more clearly correct to explicitly include the reference to the time and size in the completion object

Well, I've been running a migration script for over a week straight now that uses aio_stat(), and I'm not seeing any leaking of memory from here.

**Comment by jdurgin** [comment]
Good to know. For my own edification I verified that pmtime and psize are saved in a closure for on_complete() by reading the intermediate C code. Thanks!

**Pull request merged**